x/evm/keeper/grpc_query.go

```diff
@@ -469,43 +471,115 @@ func (k Keeper) EstimateGas(c context.Context, req *types.EthCallRequest) (*type
 // executes the given message in the provided environment. The return value will
 // be tracer dependent.
 func (k Keeper) TraceTx(c context.Context, req *types.QueryTraceTxRequest) (*types.QueryTraceTxResponse, error) {
-	// Assemble the structured logger or the JavaScript tracer
-	var (
-		tracer     vm.Tracer
-		err        error
-		resultData []byte
-	)
+	ctx := sdk.UnwrapSDKContext(c)
+	k.WithContext(ctx)
+	params := k.GetParams(ctx)
+
+	ctx.BlockHeader()
+
+	coinbase, err := k.GetCoinbaseAddress(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	ethCfg := params.ChainConfig.EthereumConfig(k.eip155ChainID)
+	signer := ethtypes.MakeSigner(ethCfg, big.NewInt(ctx.BlockHeight()))
+	result, err := k.traceTx(coinbase, signer, req.TxIndex, params, c, ctx, ethCfg, req.Msg, req.TraceConfig)
+
+	resultData, err := json.Marshal(result)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &types.QueryTraceTxResponse{
+		Data: resultData,
+	}, nil
+}
 
+func (k Keeper) TraceBlock(c context.Context, req *types.QueryTraceBlockRequest)  (*types.QueryTraceBlockResponse, error) {
```

Comment:
This function would be unnecessary. Will remove it if I decide to go with this approach